data-node/src/main/java/org/graylog/datanode/bindings/PreflightChecksBindings.java

```diff
@@ -29,6 +30,7 @@ public class PreflightChecksBindings extends AbstractModule {
     protected void configure() {
         addPreflightCheck(OpensearchConfigSync.class);
         addPreflightCheck(OpensearchBinPreflightCheck.class);
+        addPreflightCheck(DatanodeDirectoriesLockfileCheck.class);
```

Comment:
should this test run before `OpensearchConfigSync`?

Reply:
This just adds the binding to a map, so I think we can't really influence the order of the preflight checks. This would need some deeper refactoring.

Reply:
from https://google.github.io/guice/api-docs/4.1/javadoc/com/google/inject/multibindings/MapBinder.html
"The map's iteration order is consistent with the binding order" , at least if defined in the same module.